### PR TITLE
Ssh keys (astute part)

### PR DESCRIPTION
**Pull request by warpc**
Generate ssh keys and push them to every nodes (folder /var/lib/astute)

### CODE REVIEW

File: lib/astute/deployment_engine.rb [hunk 1]
```diff
@@ -15,6 +15,9 @@
 
 require 'json'
 require 'timeout'
+require 'fileutils'
+
+KEY_DIR = "/var/lib/astute"
```
Comment (yoojinl) [inline lib/astute/deployment_engine.rb:20]: Extract in settings.

File: mcagents/uploadfile.rb [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    class Uploadfile < RPC::Agent
+      
+      action 'upload' do
+        # this action is used to distribute text file from
+        # master node to all managed nodes
+
+        validate :path, :shellsafe
+        validate :content, :string
+        validate :overwrite, :boolean
+        validate :parents, :boolean
```
Comment (yoojinl) [inline mcagents/uploadfile.rb:28]: Does validations really need here if we have type definition in ddl file?

**Comment by yoojinl** [comment]
++
Stylistic comments can be fixed in next PRs.

File: lib/astute/deployment_engine.rb (hunk 1 above)
Comment (warpc) [inline lib/astute/deployment_engine.rb:20]: This data used by puppet modules and user should not change it without changing modules.

File: mcagents/uploadfile.rb (hunk 2 above)
Comment (warpc) [inline mcagents/uploadfile.rb:28]: Good point. I check that shellsafe available in DDL. I change it in next pull. Thanks!

**Pull request merged**